ydb/core/kqp/common/kqp_data_integrity_trails.h

```diff
@@ -0,0 +1,103 @@
+#pragma once
+
+#include <openssl/sha.h>
+#include <library/cpp/string_utils/base64/base64.h>
+
+#include <ydb/core/data_integrity_trails/data_integrity_trails.h>
+
+namespace NKikimr {
+namespace NDataIntegrity {
+
+inline bool ShouldBeLogged(NKikimrKqp::EQueryAction action, NKikimrKqp::EQueryType type) {
+    if (type == NKikimrKqp::QUERY_TYPE_SQL_DDL) {
+        return false;
+    }
+
+    switch (action) {
+        case NKikimrKqp::QUERY_ACTION_EXECUTE:
+        case NKikimrKqp::QUERY_ACTION_EXECUTE_PREPARED:
+        case NKikimrKqp::QUERY_ACTION_BEGIN_TX:
+        case NKikimrKqp::QUERY_ACTION_COMMIT_TX:
+        case NKikimrKqp::QUERY_ACTION_ROLLBACK_TX:
+            return true;
+        default:
+            return false;
+    }
+}
+
+// SessionActor
+inline void LogIntegrityTrails(const NKqp::TEvKqp::TEvQueryRequest::TPtr& request, const TActorContext& ctx) {
+    if (!ShouldBeLogged(request->Get()->GetAction(), request->Get()->GetType())) {
+        return;
+    }
+    
+    auto log = [](const auto& request) {
+        TStringStream ss;
+        LogKeyValue("Component", "SessionActor", ss);
+        LogKeyValue("SessionId", request->Get()->GetSessionId(), ss);
+        LogKeyValue("TraceId", request->Get()->GetTraceId(), ss);
+        LogKeyValue("Action", ToString(request->Get()->GetAction()), ss);
+        LogKeyValue("Type", ToString(request->Get()->GetType()), ss);
+
+        if (request->Get()->HasTxControl()) {
+            LogTxControl(request->Get()->GetTxControl(), ss);
+        }
+
+        std::string queryText;
```

Comment:
Values are filtered out, right? Do we need query text then?

Reply:
Actually, no. Hashed query text is useless